### ISSUE TITLE
[Bug]: CodeActAgent unaware of the tools

### ISSUE THREAD

**Issue by imkebe** [OpenDevin/OpenDevin#3287]
### Is there an existing issue for the same bug?

- [X] I have checked the troubleshooting document at https://docs.all-hands.dev/modules/usage/troubleshooting
- [X] I have checked the existing issues.

### Describe the bug

I'm trying to do anything useful with the agent. I tried with base models like :

- llama3.1-70
- phi3-medium
- gemma-27

The initial prompt is some simple one like:

> Create a fresh NextJS Project.
> 
> You have access to terminal, workspace where you should create a project files. Also browser and Jupyter iPython to plan, execute, process and gather data.

After some iterations it create a almost valid plan however it cannot execute it:

> You are right! I apologize for the confusion. I am still under development and learning how to best fulfill request
>
> Let me try again. Since I can't directly interact with your system, I will provide you with a detailed set of instructions that, if followed carefully, should result in the creation of the project files as you requested.
>
> Please follow these steps precisely:

The Agent is unaware of the tools that he can use. Few times i was able to force him by direct instructions.

### Current OpenDevin version

```bash
0.8.x
```

### Installation and Configuration

```bash
just default docker execution
```

**Comment by neubig** [comment]
Hey @imkebe , have you also tried to do the same thing with either `gpt-4o` or `claude-3-5-sonnet`? I'm wondering if it's an issue of the LM?

**Comment by imkebe** [comment]
> Hey @imkebe , have you also tried to do the same thing with either `gpt-4o` or `claude-3-5-sonnet`? I'm wondering if it's an issue of the LM?

gpt-4o is working fine.

I've simulated the completions with OpenWebUI.

Instructions are rather simple and I thought mistral-nemo, llama3.1-x, phi3 should handle such simple instructions. It turns out the either the prompt needs to be modified for open-source LLM's or they are just dumb.

I've attached Langfuse traces with full prompts and responses:

[llama3.1_70b-instruct-q6_K.pdf](https://github.com/user-attachments/files/16564369/llama3.1_70b-instruct-q6_K.pdf)
[phi3-14b-medium-128k-instruct-q6_K.pdf](https://github.com/user-attachments/files/16564370/phi3-14b-medium-128k-instruct-q6_K.pdf)
[mistral-nemo_12b-instruct-2407-fp16.pdf](https://github.com/user-attachments/files/16564371/mistral-nemo_12b-instruct-2407-fp16.pdf)

**Comment by neubig** [comment]
Awesome, thanks. We're still working on trying to make competitive open-source LMs, but it's not an easy task. We have a larger tracking issue for that here: #1085 

In the mean time, I'll close this since we have the other issue for that discussion.
